### DISCUSSION TITLE
Usage with class components

### DISCUSSION THREAD

**Discussion by WoolDoughnut310** [vercel/swr#892] [Q&A] [answered]
Although SWR has been documented as a React hook for fetching data, would there not be an option to use the `useSWR` hook (for example) within a class-based component? I am asking this because I have many class based-components within my project that utilise state and many methods - including lifecycle methods - and it would not seem reasonable for me to try and make them into functional components just for the sake of data fetching with swr.

**Comment by koba04** [comment, marked as answer]
@WoolDoughnut310 SWR is a React hook library and is intended to be used with function components, so we cannot use it in class components. https://github.com/vercel/swr/issues/175
